dashboard/app/models/unit.rb

```diff
@@ -1897,11 +1897,15 @@ def self.build_property_hash(unit_data)
     result
   end
 
-  # A unit is considered to have a matching course if there is exactly one
+  # A unit is considered to have a matching course if there is at least one
   # unit_group for this unit
   def unit_group
-    return nil if unit_group_units.length != 1
-    UnitGroup.get_from_cache(unit_group_units[0].course_id)
+    # rubocop:disable Style/ZeroLengthPredicate
+    return nil if unit_group_units.length < 1
+    # rubocop:enable Style/ZeroLengthPredicate
```

Comment:
will this work to avoid the lint issue?
```suggestion
    return nil if unit_group_units.empty?
```
If not no worries - it might be tricky because this is rails code where "length" is not exactly what it would be in vanilla ruby.

Reply:
Yeah, I thought the same thing when looking at your [proof of concept](https://github.com/code-dot-org/code-dot-org/pull/59371/commits/20db3bf72c221e3ae11f886985b9f589aae0f42f). But, if we use `unit_group_units.empty?`, I guess it triggers another database query and it was failing tests.

Reply:
ok, sounds good to leave as-is then